components/src/dynamo/vllm/handlers.py

```
@@ -825,6 +843,450 @@ async def stop_profile(self, body: dict) -> dict:
+            return {"status": "error", "message": "Missing 'model_path' in body"}
+        version = body.get("weight_version", "unknown")
+        rpc = body.get("engine_rpc", "reload_weights")
+        kwargs = {"weights_path": path} if rpc == "reload_weights" else {"weight_path": path}
+        try:
+            await self.engine_client.collective_rpc(rpc, kwargs=kwargs)
+            self._weight_version = version
+            logger.info(f"[RL] Weights loaded from {path} (version={version}, rpc={rpc})")
+            return {"status": "ok", "version": version}
+        except EngineDeadError as e:
+            self._shutdown_on_engine_dead(e)
+        except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)